nncf/common/pruning/weights_flops_calculator.py

```
@@ -106,35 +110,45 @@ def count_flops_and_weights_per_node(
+            num_out_channels = output_channels.get(name, layer_attr.out_channels)
+            kernel_size = kernel_sizes.get(name, layer_attr.kernel_size)
             if is_prunable_depthwise_conv(node):
                 # Prunable depthwise conv processed in special way
                 # because common way to calculate filters per
                 # channel for such layer leads to zero in case
                 # some of the output channels are pruned.
                 filters_per_channel = 1
             else:
-                filters_per_channel = num_out_channels // node.layer_attributes.groups
+                filters_per_channel = num_out_channels // layer_attr.groups
 
             flops_numpy = (
-                2 * np.prod(kernel_size) * num_in_channels * filters_per_channel * np.prod(output_shapes[name])
+                2
+                * int(np.prod(kernel_size))
+                * num_in_channels
+                * filters_per_channel
+                * int(np.prod(output_shapes[name]))
             )
-            weights_numpy = np.prod(kernel_size) * num_in_channels * filters_per_channel
+            weights_numpy = int(np.prod(kernel_size)) * num_in_channels * filters_per_channel
 
-            flops[name] = flops_numpy.astype(int).item()
-            weights[name] = weights_numpy.astype(int).item()
+            flops[name] = flops_numpy
+            weights[name] = weights_numpy
 
         for node in graph.get_nodes_by_metatypes(self._linear_op_metatypes):
             name = node.node_name
             if name in op_addresses_to_skip:
                 continue
 
-            num_in_features = input_channels.get(name, node.layer_attributes.in_features)
-            num_out_features = output_channels.get(name, node.layer_attributes.out_features)
+            layer_attr = node.layer_attributes
+            if not isinstance(layer_attr, LinearLayerAttributes):
+                continue
```

Let's raise an internal error here

---

done